qutebrowser/config/config.py

```diff
@@ -731,28 +732,49 @@ def set_command(self, win_id, section_=None, option=None, value=None,
             print_ = True
         else:
             with self._handle_config_error():
-                if option.endswith('!') and option != '!' and value is None:
+                if option.endswith('!') and option != '!' and len(values) == 0:
+                    # Handle inversion as special cases of the cycle code path
                     option = option[:-1]
                     val = self.get(section_, option)
-                    layer = 'temp' if temp else 'conf'
                     if isinstance(val, bool):
-                        self.set(layer, section_, option, str(not val).lower())
+                        values = ('false', 'true')
```

Comment:
I'd prefer a list here ([this is why](http://stackoverflow.com/a/626871/2085149))